typescript/common.ts

```diff
@@ -1,5 +1,5 @@
 // From https://github.com/microsoft/TypeScript/issues/1897
-export type JSONPrimitive = string | number | boolean | null | unknown;
+export type JSONPrimitive = string | number | boolean | null | undefined;
```

Comment:
I think we should revisit later and see if we can remove undefined in favour of making the property optional where applicable. `undefined` is not actually a valid JSON value if we were to deserialize a JSON string

Reply:
undefined is definitely better than unknown though, right ?